test/rand.jl

```diff
@@ -47,6 +49,10 @@ for (f,T) in ((rand!,Int64),),
     f(A)
 end
 
+@test_throws ErrorException CURAND.normal_rng(CuArray{Cuint}(undef, 10))
```

Comment:
No need to check these using the internal API, you can trigger the exception using exported functions:

```
julia> CURAND.rand_poisson!(CuArray{Float64}(undef, 10))
ERROR: CuArrays does not support generating Poisson distributed numbers of type Float64
```